apps/web/src/pages/integrations/IntegrationsStorePage.tsx

```diff
@@ -22,14 +22,12 @@ import { useIntegrations } from '../../hooks';
 import { When } from '../../components/utils/When';
 import { NovuEmailProviderModal } from './components/NovuEmailProviderModal';
 import { NovuInAppProviderModal } from './components/NovuInAppProviderModal';
+import { useProviders } from './useProviders';
 import { NovuSmsProviderModal } from './components/NovuSmsProviderModal';
 
 export function IntegrationsStore() {
-  const { integrations, loading: isLoading, refetch } = useIntegrations();
-  const [emailProviders, setEmailProviders] = useState<IIntegratedProvider[]>([]);
-  const [smsProvider, setSmsProvider] = useState<IIntegratedProvider[]>([]);
-  const [chatProvider, setChatProvider] = useState<IIntegratedProvider[]>([]);
-  const [pushProvider, setPushProvider] = useState<IIntegratedProvider[]>([]);
+  const { loading: isLoading, refetch } = useIntegrations();
+  const { emailProviders, smsProvider, chatProvider, pushProvider, inAppProvider } = useProviders();
```

Comment:
I like it like this because it is obvious what we refetch and is loading here...

Reply:
you fetch integrations but use providers when it's not loading, to me it's not obvious... IMO I think it's a waste of React resources, it might be a bad thing if it's going to be a habit...

Reply:
@LetItRock what do you mean with waste of React resources?
Regardless the discussion here, I will always favour the consistency so maybe is something worth for tech summit (I know I missed the last one 😓 ).

Reply:
I do not have a strong oppinon here so will fix this...

Reply:
@p-fernandez the `useIntegrations()` is used inside the `useProviders()` hook, so declaring both here instead of just `const { loading: isLoading, refetch, ... } = useProviders();` is a waste of resources because React has to create the `useIntegrations` hook state and do all it's logic to fulfill that state, and it's not that simple because there is a lot of going on inside the core of `react-query`.

Reply:
Ah now I understand what you meant with waste of resources, as we are duplicating the calls. I do agree with that bit, indeed. 👍🏻